### PR TITLE
Feature/stepper motor

### PR DESCRIPTION
**Pull request by leahkiner**
* **Tickets addressed:** None
* **Review:** By commit  by commit
* **Merge strategy:** merge (no squash)

## Description
This stepper motor controller module calculates the number of steps required to actuate from an initial stepper motor angle to a final reference angle. The input message for this module is a `hingedRigidBody` reference angle and the output is the calculated number of steps in a new `motorStepCommand` message. The motor angles are discretized by a constant step angle; therefore the motor cannot actuate to any desired angle. The desired angle is adjusted to the nearest multiple of the motor step angle for cases where the desired angle is not a valid motor angle.

## Verification
There are two unit tests included in this PR. The first tests the module for both positive and negative steps. The second test script checks that the module correctly computes the required number of steps for new messages that interrupt the motor actuation.

## Documentation
No documentation is added currently.

## Future work
Documentation for this module will be added in the future.

### CODE REVIEW

File: src/architecture/msgPayloadDefC/MotorStepCountMsgPayload.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+#ifndef motorStepCountSimMsg_h
+#define motorStepCountSimMsg_h
+
+
+ /*! @brief Structure containing number of commandaed stepper motor steps */
```
Comment (patkenneally) [inline src/architecture/msgPayloadDefC/MotorStepCountMsgPayload.h]: :large_orange_diamond: spelling of "commanded"

File: src/fswAlgorithms/effectorInterfaces/stepperMotor/stepperMotor.c [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+/*! This method performs a complete reset of the module. The input messages are checked to ensure they are linked.
+ @return void
+ @param configData The configuration data associated with the module
+ @param callTime [ns] Time the method is called
+ @param moduleID The module identifier
+*/
+void Reset_stepperMotor(StepperMotorConfig* configData, uint64_t callTime, int64_t moduleID) {
+    // Check if the required input message is linked
+    if (!HingedRigidBodyMsg_C_isLinked(&configData->spinningBodyInMsg)) {
+        _bskLog(configData->bskLogger, BSK_ERROR, "Error: stepperMotor.spinningBodyInMsg wasn't connected.");
```
Comment (patkenneally) [inline src/fswAlgorithms/effectorInterfaces/stepperMotor/stepperMotor.c]: :small_red_triangle_down: You can remove "Error: " from this string as it's taken care of by the logging call.

File: src/fswAlgorithms/effectorInterfaces/stepperMotor/_UnitTest/test_stepperMotor.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+# stand-along python script
+if __name__ == "__main__":
+    stepperMotorTestFunction(
+                 False,
+                 1 * (np.pi / 180),     # stepAngle
+                 1,                     # stepTime
+                 0,                     # initialAngle
+                 10 * (np.pi / 180),    # desiredAngle
+               )
+
```
Comment (patkenneally) [inline src/fswAlgorithms/effectorInterfaces/stepperMotor/_UnitTest/test_stepperMotor.py]: File is looking for a new line at the end.

**Review by patkenneally — APPROVED** [review]
I can merge this upon the small adjustments. Nice work!

File: src/architecture/msgPayloadDefC/MotorStepCountMsgPayload.h (hunk 1 above)
Comment (leahkiner) [reply src/architecture/msgPayloadDefC/MotorStepCountMsgPayload.h]: Done

File: src/fswAlgorithms/effectorInterfaces/stepperMotor/stepperMotor.c (hunk 2 above)
Comment (leahkiner) [reply src/fswAlgorithms/effectorInterfaces/stepperMotor/stepperMotor.c]: Done

File: src/fswAlgorithms/effectorInterfaces/stepperMotor/_UnitTest/test_stepperMotor.py (hunk 3 above)
Comment (leahkiner) [reply src/fswAlgorithms/effectorInterfaces/stepperMotor/_UnitTest/test_stepperMotor.py]: I think I fixed this..?

Comment (patkenneally) [reply src/fswAlgorithms/effectorInterfaces/stepperMotor/_UnitTest/test_stepperMotor.py]: yes

**Pull request merged**